src/framework/script/script-type.js

```diff
@@ -6,6 +6,17 @@ import { ScriptAttributes } from './script-attributes.js';
 
 const funcNameRegex = new RegExp('^\\s*function(?:\\s|\\s*\\/\\*.*\\*\\/\\s*)+([^\\(\\s\\/]*)\\s*');
 
+/**
+ * EventListener object used for storing what events on EventHandlers the ScriptType is listening for.
+ *
+ * @typedef {object} EventListener
```

Comment:
`@typedef` is too good to be allowed to use, you need to remove it and mess around with `any` instead. :wink:

Reply:
I did wonder if I should use a typedef or a new class definition 🤔

Reply:
Proper way by example: https://github.com/playcanvas/engine/blob/6d4766e7dd7cdc2e4ecfac439e4cb4b85b2e0404/src/framework/anim/state-graph/anim-state-graph.js#L17

(while removing neat IntelliSense)